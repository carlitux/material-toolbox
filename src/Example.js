// @flow
import * as React from 'react';

import TabBar from './material-toolbox/tabs/Bar';
import Tab from './material-toolbox/tabs/Tab';
import Text from './material-toolbox/typography/Text';

import Highlighter from './Highlighter';

type Props = {
  title: string,
  children: React.Node,
  source: string,
  scss?: string,
};

type State = {
  tabSelected: number,
};

export default class Example extends React.PureComponent<Props, State> {
  state = {
    tabSelected: 0,
  };

  handleChange = (index: number) => {
    this.setState({ tabSelected: index });
  };

  render() {
    const { title, children, source, scss } = this.props;

    return (
      <div className="example">
        <Text textStyle="title" component="h2">
          {title}
        </Text>
        <TabBar onChange={this.handleChange} refreshAt={1800}>
          <Tab>Example</Tab>
          <Tab>Source</Tab>
        </TabBar>
        <section className={this.state.tabSelected === 0 ? '' : 'hide'}>
          {children}
        </section>
        <section className={this.state.tabSelected === 1 ? '' : 'hide'}>
          <Highlighter language="jsx">{source}</Highlighter>
          {scss && <Highlighter language="scss">{scss}</Highlighter>}
        </section>
      </div>
    );
  }
}
