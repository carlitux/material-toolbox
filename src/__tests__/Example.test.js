// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Example from '../Example';

describe('component::Example', () => {
  it('Should match snapshoot', () => {
    const component = shallow(
      <Example title="Example title" source="source code">
        source code
      </Example>,
    );
    expect(component).toMatchSnapshot();
  });

  it('Should match snapshoot scss', () => {
    const component = shallow(
      <Example title="Example title" source="source code" scss="scss code">
        source code
      </Example>,
    );
    expect(component).toMatchSnapshot();
  });
});
