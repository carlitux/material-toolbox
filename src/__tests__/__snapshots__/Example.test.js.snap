// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::Example Should match snapshoot 1`] = `
<div
  className="example"
>
  <Text
    adjustMargin={false}
    component="h2"
    textStyle="title"
  >
    Example title
  </Text>
  <TabBar
    cssOnly={false}
    onChange={[Function]}
    refreshAt={1800}
  >
    <Tab
      active={false}
      cssOnly_={false}
    >
      Example
    </Tab>
    <Tab
      active={false}
      cssOnly_={false}
    >
      Source
    </Tab>
  </TabBar>
  <section
    className=""
  >
    source code
  </section>
  <section
    className="hide"
  >
    <Highlighter
      language="jsx"
    >
      source code
    </Highlighter>
  </section>
</div>
`;

exports[`component::Example Should match snapshoot scss 1`] = `
<div
  className="example"
>
  <Text
    adjustMargin={false}
    component="h2"
    textStyle="title"
  >
    Example title
  </Text>
  <TabBar
    cssOnly={false}
    onChange={[Function]}
    refreshAt={1800}
  >
    <Tab
      active={false}
      cssOnly_={false}
    >
      Example
    </Tab>
    <Tab
      active={false}
      cssOnly_={false}
    >
      Source
    </Tab>
  </TabBar>
  <section
    className=""
  >
    source code
  </section>
  <section
    className="hide"
  >
    <Highlighter
      language="jsx"
    >
      source code
    </Highlighter>
    <Highlighter
      language="scss"
    >
      scss code
    </Highlighter>
  </section>
</div>
`;
